features/step_definitions/session_steps.rb

```diff
@@ -57,7 +57,7 @@
   else
     try_buyer_login_internal(username, password)
   end
-  assert_current_user(username)
+  assert user_is_logged_in(username)
```

Comment:
All these assertions are to avoid a situation where "Sign in" is clicked, but the request hasn't completed yet, and a subsequent page access redirects to login again.